### PR TITLE
Add a check for use of temporary tensors

### PR DESCRIPTION
**Pull request by zdevito**
Addresses #45

### CODE REVIEW

File: include/tc/lang/sema.h [hunk 1]
```diff
@@ -601,5 +613,7 @@ struct Sema {
   // if you write to an input, using it in a range expression is no longer
   // allowed
   std::unordered_set<std::string> live_input_names;
+
+  std::unordered_set<std::string> non_temporaries;
```
Comment (prigoyal) [inline include/tc/lang/sema.h]: can you make it cameCase i.e. `nonTemporaries` ? this is as per our coding conventions so we should follow them as well :)

**Comment by zdevito** [comment]
Are we working off a different convention than what is is CodingConvensions.md? Because it says either is ok? I changed it to lowerCamelCase as requested.

```
### Class data members ###

As with variables, use `lowerCamelCase` or `lower_case_with_underscores`.
```

**Comment by prigoyal** [comment]
we are using `lowerCamelCase`, you can look at other code reviews and most of our code except lang/ , we are trying to strictly follow this. I'll clean up the underscores part , that shouldn't be there

**Comment by prigoyal** [comment]
looks good, thanks much!

**Comment by skimo-openhub** [comment]
On Fri, Feb 23, 2018 at 05:13:44AM -0800, Priya Goyal wrote:
> we are using `lowerCamelCase`, you can look at other code reviews and most of our code except lang/ , we are trying to strictly follow this. I'll clean up the underscores part , that shouldn't be there

If this is a requirement now for data members,
then please update the relevant section in CodingConvensions.md:

On Thu, Feb 22, 2018 at 09:45:34PM -0800, Zachary DeVito wrote:
> Are we working off a different convention than what is is CodingConvensions.md? Because it says either is ok? I changed it to lowerCamelCase as requested.
> 
> ```
> ### Class data members ###
> 
> As with variables, use `lowerCamelCase` or `lower_case_with_underscores`.
> ```

skimo

**Pull request merged**